docs/docs/reference/other-new-features/targetName.md

```diff
@@ -0,0 +1,79 @@
+```
+The `@targetName` annotation has no bearing on Scala usages. Any application of that method in Scala has to use `++=`, not `append`.
+
+### Details
+
+ 1. `@targetName` is defined in package `scala.annotation`. It takes a single argument
+    of type `String`. That string is called the _external name_ of the definition
+    that's annotated.
+
+ 2. A `@targetName` annotation can be given for all kinds of definitions.
+
+ 3. The name given in a `@targetName` annotation must be a legal name
+    for the defined entities on the host platform.
+
+ 4. It is recommended that definitions with symbolic names have a `@targetName` annotation. This will establish an alternate name that is easier to search for and
+ will avoid cryptic encodings in runtime diagnostics.
+
+ 5. Definitions with names in backticks that are not legal host platform names
+    should also have a `@targetName` annotation.
+
+### Relationship with Overriding
+
+`@targetName` annotations are significant for matching two method definitions to decide whether they conflict or override each other. Two method definitions match if they have the same name, signature, and erased name. Here,
+
+ - The _signature_ of a definition consists of the names of the erased types of all (value-) parameters and the method's result type.
+ - The _erased name_ of a method definition is its target name if a `@targetName`
+   annotation is given and its defined name otherwise.
+
+This means that `@targetName` annotations can be used to disambiguate two method definitions that would otherwise clash. For instance.
+```scala
+def f(x: => String): Int = x.length
+def f(x: => Int): Int = x + 1  // error: double definition
+```
+The two definitions above clash since their erased parameter types are both `Function0`, which is the type of the translation of a by-name-parameter. Hence
+they have the same names and signatures. But we can avoid the clash by adding  a `@targetName` annotation to either method or to both of them. E.g.
+```scala
+@targetName("f_string")
+def f(x: => String): Int = x.length
+def f(x: => Int): Int = x + 1  // OK
+```
+This will produce methods `f_string` and `f` in the generated code.
+
+As usual, any overriding relationship in the generated code must also
+be present in the original code. So the following example would be in error:
+```scala
+import annotation.targetName
+class A:
+  def f(): Int = 1
+class B extends A:
+  targetName("f") def g(): Int = 2
+```
+Here, the original methods `g` and `f` do not override each other since they have
+different names. But once we switch to target names, there is a clash that is reported by the compiler:
+```
+-- [E120] Naming Error: test.scala:4:6 -----------------------------------------
+4 |class B extends A:
+  |      ^
+  |      Name clash between defined and inherited member:
+  |      def f(): Int in class A at line 3 and
+  |      def g(): Int in class B at line 5
+  |      have the same name and type after erasure.
```

Comment:
Regarding the error message, maybe it's worth proposing the user the option of using `@targetName` to resolve the error.